### PR TITLE
chore: Migrate release builds to GitHub Actions WPB-3505

### PR DESCRIPTION
**Pull request by SzymonMrozek**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-3505" title="WPB-3505" target="_blank">TaskWPB-3505</a>  Migrate release builds to GitHub Actions
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Solutions

Migrate from CirrusCI to GithubActions with [CirrusCI runners](https://github.com/apps/cirrus-runners)

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: .github/workflows/app_release.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
+on:
```
Comment (netbe) [inline .github/workflows/app_release.yml:1]: so this workflow can be triggered directly alone?

File: .github/workflows/bund_app_release_restricted.yml [hunk 2]
```diff
@@ -7,35 +7,13 @@ permissions:
...
-          fetch-depth: 0
-      - name: Set up Ruby
-        uses: ruby/setup-ruby@v1
-        with:
-          ruby-version: '3.1'
-      - name: Install dependencies
-        run: bundle install
-      - name: Trigger build
-        run: bundle exec fastlane trigger_api_build_cirrus_ci task:"AppStore C1 Restricted"
+    uses: ./.github/workflows/app_release.yml
```
Comment (netbe) [inline .github/workflows/bund_app_release_restricted.yml]: does this solve the issue with one would be cancelled automatically by the other?

**Review by netbe — APPROVED** [review]
That simplifies everything :)

1. Do I understand that tests (run_tests action) would still be performed on github runners and not cirrus ci ones?

2. Do you have an output job how it looks like?

File: .github/workflows/bund_app_release_restricted.yml (hunk 2 above)
Comment (SzymonMrozek) [reply .github/workflows/bund_app_release_restricted.yml]: yes

File: .github/workflows/app_release.yml (hunk 1 above)
Comment (SzymonMrozek) [reply .github/workflows/app_release.yml:1]: it's reusable workflow, wouldn't be triggered manually but called by other workflows

**Comment by SzymonMrozek** [comment]
> 

1. Yes, let's first verify our performance on cirrus runners, later on maybe we could switch run-tests too... but I don't think it's necessary right now 
2. https://github.com/wireapp/wire-ios-mono/actions/runs/5660583147

File: .github/workflows/app_release.yml (hunk 1 above)
Comment (netbe) [reply .github/workflows/app_release.yml:1]: will this be listed though?
maybe we should move it to .github/workflows/templates/ to make it more obvious

**Comment by netbe** [comment]
@SzymonMrozek ok so we can add notifications to wire now when build is triggered/finished, should we added it now ?

**Comment by SzymonMrozek** [comment]
> @SzymonMrozek ok so we can add notifications to wire now when build is triggered/finished, should we added it now ?

No it won't be listed, but yes I could add that to templates

**Comment by SzymonMrozek** [comment]
> @SzymonMrozek ok so we can add notifications to wire now when build is triggered/finished, should we added it now ?

you mean notifications to Wire app? If it's needed then I would suggest making new PR with it

File: .github/workflows/app_release.yml (hunk 1 above)
Comment (SzymonMrozek) [reply .github/workflows/app_release.yml:1]: actually it cannot be 
`invalid value workflow reference: workflows must be defined at the top level of the .github/workflows/ directory`

**Review by David-Henner — APPROVED** [review]

**Pull request merged**